integration_test.go

```diff
@@ -375,10 +375,24 @@ func TestStop(t *testing.T) {
 }
 
 func writeDataIntoTestapp(t *testing.T, data string) {
```

Comment:
A better factoring would be to name this function `atomicWriteFile(t, data, filename)` Then it's not too tightly coupled with just writing a single file into the testapp.

Up to you if you want to make this change before commiting